compiler/rustc_span/src/hygiene.rs

```diff
@@ -82,16 +82,16 @@ impl SyntaxContextData {
             parent: SyntaxContext::root(),
             opaque: SyntaxContext::root(),
             opaque_and_semitransparent: SyntaxContext::root(),
-            dollar_crate_name: kw::DollarCrate,
+            dollar_crate_name: Some(kw::DollarCrate),
         }
     }
 
     fn decode_placeholder() -> SyntaxContextData {
-        SyntaxContextData { dollar_crate_name: kw::Empty, ..SyntaxContextData::root() }
+        SyntaxContextData { dollar_crate_name: None, ..SyntaxContextData::root() }
     }
 
     fn is_decode_placeholder(&self) -> bool {
-        self.dollar_crate_name == kw::Empty
```

Comment:
This is not a placeholder for `dollar_crate_name`, it means that the whole `SyntaxContextData` is a placeholder and will never be used (there's a bunch of `debug_assert`s in this file checking for it).

I think it's ok to use `sym::dummy` here, but there's an ongoing work (#129827) to eliminate these placeholders entirely, so it's better not to change this otherwise.

Reply:
Ok.